topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -102,16 +103,20 @@ public void handle(Long inflightTimerId) {
                     if (zkResult.failed()) {
                         LOGGER.warn("Error disconnecting from zookeeper: {}", String.valueOf(zkResult.cause()));
                     }
-                    LOGGER.debug("Closing AdminClient {}", adminClient);
-                    adminClient.close(deadline - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
-
-                    HttpServer healthServer = this.healthServer;
-                    if (healthServer != null) {
-                        healthServer.close();
+                    long timeoutMs = Math.max(1, deadline - System.currentTimeMillis());
```

Comment:
why this adjustment? did you notice any strange `deadline - System.currentTimeMillis()` value?

Reply:
I observed an exception caused by a negative timeout, and I know from experience that a 0 argument also causes an exception.